lib/android/nit_activity.nit

```diff
@@ -0,0 +1,217 @@
+	#include <android/log.h>
+
+	// Nit's App running instance
+	App global_app;
+
+	// Java VM that launched this program
+	JavaVM *global_jvm;
+
+	// JNI callback on loading this program
+	jint JNI_OnLoad(JavaVM *vm, void *reserved) {
+		// Set aside the Java VM
+		global_jvm = vm;
+
+		// Invoke Nit system and main
+		main(0, NULL);
+
+		return JNI_VERSION_1_2;
+	}
+
+	/*
+	 * Implementations of NitActivity.java native methods
+	 */
+
+	JNIEXPORT jint JNICALL Java_nit_app_NitActivity_nitRegisterActivity
+	  (JNIEnv *env, jobject java_activity)
+	{
+		Activity nit_activity = App_register_activity(global_app, java_activity);
+		Activity_incr_ref(nit_activity);
+		return (jint)(void*)nit_activity;
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnCreate
+	  (JNIEnv *env, jobject java_activity, jint nit_activity, jobject saved_state)
+	{
+		Activity_on_create((Activity)nit_activity, saved_state);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnStart
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_start((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnResume
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_resume((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnPause
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_pause((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnStop
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_stop((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnDestroy
+	  (JNIEnv *env, jobject java_activity, jint nit_activity)
+	{
+		Activity_on_destroy((Activity)nit_activity);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnSaveInstanceState
+	  (JNIEnv *env, jobject java_activity, jint nit_activity, jobject saved_state)
+	{
+		Activity_on_save_instance_state((Activity)nit_activity, saved_state);
+	}
+
+	JNIEXPORT void JNICALL Java_nit_app_NitActivity_nitOnRestoreInstanceState
+	  (JNIEnv *env, jobject java_activity, jint nit_activity, jobject saved_state)
+	{
+		Activity_on_restore_instance_state((Activity)nit_activity, saved_state);
+	}
+`}
+
+# Wrapper to our Java `NitActivity`
+extern class NativeNitActivity in "Java" `{ nit.app.NitActivity `}
+	super NativeActivity
+end
+
+redef class Sys
+	redef fun jvm `{ return global_jvm; `}
+end
+
+redef class App
+	# Known activities
+	var activities = new Array[Activity]
+
+	# The main Java Activity of this application
+	redef fun native_activity do return activities.first.native
+
+	redef fun setup do set_global_app
+
+	# Register app in C
+	private fun set_global_app import register_activity,
+	Activity.on_create, Activity.on_start, Activity.on_pause, Activity.on_resume,
+	Activity.on_destroy, Activity.on_stop,
+	Activity.on_save_instance_state, Activity.on_restore_instance_state `{
```

Comment:
They are use as callbacks from Java/C. This method enables these callbacks by setting aside the receiver in C, so we declare the importations here.